plugins/messagelayer/consensus_plugin.go

```diff
@@ -459,9 +461,19 @@ func makeStatement(roundStats *vote.RoundStats) {
 			)
 		default:
 		}
+		// limit the size of statements
+		maxSize := payload.MaxSize
+		if (len(conflicts)*statement.ConflictLength + len(timestamps)*statement.TimestampLength) >= int(0.9*float64(maxSize)) {
+			broadcastStatement(conflicts, timestamps)
```

Comment:
You should replace this if with a `while` loop that keeps sending sub-slices of `conflicts` and `timestamps` until you run out of elements in both slices. 
Currently this `if` statement doesn't split the conflicts into multiple statements.

Reply:
It shouldn't split it into separate statements. Every time the counter reaches the max size of a statement it will send a statement and reset the counters. 

But your approach would also work. Does it have advantages over this one?

Reply:
Sorry! - I thought it's outside of the `for` loop. It's ok then :)